File: app/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/fragments/investigation/mapsmenu/mapdisplay/data/models/MapListModel.java
package com.TritiumGaming.phasmophobiaevidencepicker.activities.fragments.investigation.mapsmenu.mapdisplay.data.models;

import com.TritiumGaming.phasmophobiaevidencepicker.activities.fragments.investigation.mapsmenu.mapdisplay.io.models.WorldMapWrapper;

import java.util.ArrayList;
import java.util.Map;

public class MapListModel {

    private int currentMapID = 0;
    public ArrayList<MapModel> mapModels;

    public MapListModel(WorldMapWrapper worldMapWrapper) {
        mapModels = new ArrayList<>();

        for(Map<String, WorldMapWrapper.WorldMap> wm: worldMapWrapper.maps) {
            WorldMapWrapper.WorldMap worldMap = wm.get("map_data");
            if(worldMap != null) {
                mapModels.add(new MapModel(worldMap));
            }
        }
    }

    public MapModel getMapById(int id) {
        for(MapModel m: mapModels) {
            if(m.mapId == id) {
                return m;
            }
        }

        return null;
    }

    public MapModel getCurrentMap() {
        if(currentMapID >= mapModels.size())
            return new MapModel();

        return mapModels.get(currentMapID);
    }

    public void setCurrentMap(int selectedIndex) {
        if(selectedIndex >= mapModels.size()) {
            selectedIndex = Math.max(mapModels.size()-1, 0);
        }

        currentMapID = selectedIndex;
    }

    public synchronized void print() {
        for (MapModel m : mapModels) {
            m.print();
        }
    }

    public void orderRooms() {
        for(MapModel m: mapModels) {
            m.orderRooms();
        }
    }
}

File: app-compose/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/operation/domain/map/complex/model/worldmaps/MapListModel.kt
package com.tritiumgaming.phasmophobiaevidencepicker.operation.domain.map.complex.model.worldmaps

class MapListModel() {

    private var mapModels: ArrayList<MapModel> = ArrayList()

    fun addMapModel(model: MapModel) {
        mapModels.add(model)
    }

    val shortenedMapNames: ArrayList<String>
        get() {
            val names = ArrayList<String>(mapModels.size)
            for (mapModel in mapModels) {
                names.add(mapModel.mapNameShort)
            }

            return names
        }

    fun getMapById(id: String): MapModel? {
        for (m in mapModels) {
            if (m.mapId == id) {
                return m
            }
        }

        return null
    }

    @Synchronized
    fun print() {
        for (m in mapModels) {
            m.print()
        }
    }

    fun orderRooms() {
        for (m in mapModels) {
            m.orderRooms()
        }
    }

}
